spirv_cross/src/msl.rs

```diff
@@ -49,6 +49,33 @@ pub struct ResourceBinding {
     pub force_used: bool,
 }
 
+/// A HLSL shader model version.
+#[allow(non_snake_case, non_camel_case_types)]
+#[derive(Copy, Clone, Debug, Hash, Eq, PartialEq)]
+pub enum Version {
+    ios_1_0,
```

Comment:
Should we keep the platform and MSL version separate to match SPIRV-Cross and to avoid repeating the MSL versions per platform? I was thinking we could just have `enum Platform { iOS, macOS }` and ` enum Version { V1_0, V1_1, V1_2, V2_0 }`

Reply:
Yes, it's an option. However, I could see one counter-point. The whole idea of using an enum as opposed to raw version numbers is to restrict the user to what's available and what's not.. MSL spec clearly states only the given combinations available, so allowing the user to do something like (macOs, V1_0) would be worse than the current approach.
Overall though, I do agree we should separate. I'll do that now.